src/fhir_populator/populator.py

```diff
@@ -424,9 +442,79 @@ def get_latest_package_version(self, package_name: str) -> str:
         self.log.debug(f"Latest version: {last_version}")
         return last_version
 
+    def persistate(self, dependency_graph):
+        def copy_directory_structure(src, dst):
+            """
+            Recursively creates directories from src to dst
+            """
+            for root, dirs, files in os.walk(src):
+                for directory in dirs:
+                    src_dir = os.path.join(root, directory)
+                    dst_dir = os.path.join(dst, os.path.relpath(src_dir, src))
+                    if not os.path.exists(dst_dir):
+                        os.makedirs(dst_dir)
+
+        def copy_files_with_structure(src, dst):
+            """
+            Copies files along with the directory structure from src to dst
+            """
+            copy_directory_structure(src, dst)
+            for root, dirs, files in os.walk(src):
+                for file in files:
+                    src_file = os.path.join(root, file)
+                    dst_file = os.path.join(dst, os.path.relpath(src_file, src))
+                    shutil.copy2(src_file, dst_file)
+
+        persistence_dir = self.args.persistence_dir
+        # copy files to persistence_dir from extract_dir
+        copy_files_with_structure(self.extract_dir, persistence_dir)
+
+        # update the dependency graph with the new paths
+        for node in dependency_graph.nodes:
+            node_path = dependency_graph.nodes[node]["path"]
+            node_path = node_path.replace(self.extract_dir, "")
+            # remove leading slash if present
+            if node_path.startswith("/") or node_path.startswith("\\"):
+                node_path = node_path[1:]
+            elif node_path.startswith(r"\\") or node_path.startswith(r"//"):
+                node_path = node_path[2:]
+            dependency_graph.nodes[node]["path"] = node_path
+
+        # dump the dependency graph to a file in persistence_dir
+        if persistence_dir is not None:
+            graph_path = os.path.join(persistence_dir, "dependency_graph.json")
+            with open(graph_path, "w") as graph_fs:
+                json.dump(nx.node_link_data(dependency_graph), graph_fs)
+
+    def load_persisted(self):
+        persistence_dir = self.args.persistence_dir
+        graph_path = os.path.join(persistence_dir, "dependency_graph.json")
+        with open(graph_path, "r") as graph_fs:
+            dependency_graph = nx.node_link_graph(json.load(graph_fs))
+            # update the dependency graph with the new paths
+            for node in dependency_graph.nodes:
+                node_path = dependency_graph.nodes[node]["path"]
+                node_path = os.path.join(persistence_dir, node_path)
+                dependency_graph.nodes[node]["path"] = node_path
+        return dependency_graph
+
     def populate(self) -> None:
-        packages = self.resolve_package_versions()
-        dependency_graph = self.download_packages(packages)
+        packages = None
```

Comment:
do you see a use case where you restore from persistence, but don't want everything? If you use the same persistence dir for multiple calls (for whatever reason) you currently would restore all packages. Currently, the `package` argument is ignored for the call to `self.load_persisted`, I don't think that's ideal.